### ISSUE TITLE
Undocumented supervisor dependency

### ISSUE THREAD

**Issue by Kolano** [cuckoosandbox/cuckoo#1403]
The Cuckoo startup documentation...
https://cuckoo.sh/docs/usage/start.html#cuckoo-in-the-background
...outlines the supervisor script now provided with Cuckoo, but the installation documentation fails to list supervisor as a dependency.

sudo apt-get install supervisor

**Comment by Kolano** [comment]
Some explanation may also be needed on how to configure supervisor to restart Cuckoo post a reboot. I presume one would need to migrate the supervisord.conf file to...
/etc/supervisor/conf.d
...but not really sure. It seems the current instructions require rerunning...
supervisord -c $CWD/supervisord.conf
...post every reboot.

**Comment by jbremer** [comment]
Yes, that sounds about right. If you have suggestions on how to automate that, let me know. Perhaps `supervisor` has come up with a good solution to this in the meanwhile?

**Comment by Kolano** [comment]
So it seems the default install for Supervisor is a mess regarding file permissions, Refer [Issue 173](https://github.com/Supervisor/supervisor/issues/173). Beyond the socket permissions issue there is also permissions issues with non-root access to the /var/log/supervisor directory. Commands below account for such (note this install is not using the "cuckoo" user, which may require alternate permission changes)...

```
echo -e "Account for supervisor.sock 'Permission denied' errors" #https://github.com/Supervisor/supervisor/issues/173
sudo chmod 766 /var/run/supervisor.sock
sudo sed -i 's/chmod=0700/chmod=0766/' /etc/supervisor/supervisord.conf #Make supervisor.sock file universally writable

echo -e "\nAccount for /var/log/supervisor 'Permission denied' errors" 
chown cuckoo /var/log/supervisor #This may need to specify $USER if not using the cuckoo user, and some alternate handling to preserve log output by all possibly supervisor users: cuckoo / non-cuckoo / root)
```

With permissions issues addressed we can add a "cuckoo.conf" file to Supervisors's default config location "/etc/supervisor/conf.d"...
```
[program:cuckoo-daemon]
command = cuckoo -d -m 10000
user = cuckoo
startsecs = 30
autorestart = true

[program:cuckoo-process]
command = cuckoo process p%(process_num)d
process_name = cuckoo-process_%(process_num)d
numprocs = 4
user = cuckoo
autorestart = true

[group:cuckoo]
programs = cuckoo-daemon, cuckoo-process

[program:distributed]
command = python -m cuckoo.distributed.worker
user = cuckoo
autostart = false
autorestart = true
environment = CUCKOO_APP="worker",CUCKOO_CWD="/home/user/.cuckoo"
```
... **NOTE: the Cuckoo Working Directory location will need to be appropriately updated**

With that Cuckoo will now run with the default Supervisor process started on boot, though I'll probably also want to migrate my Upstart handlers for cuckoo-root / cuckoo-web to be handled via Supervisor as well; might be nice if default configs for such were generated

**Comment by Kolano** [comment]
It would also be good to revise the name of the "distributed" process to more clearly associate it with Cuckoo.

**Comment by jbremer** [comment]
`cuckoo web` isn't in `supervisord.conf` because a better setup for that would be deploying it with `uwsgi`+`nginx` for which we have documentation as well :-)
